runtimes/server-interface/workspace.ts

```diff
@@ -19,15 +19,25 @@ export type Workspace = {
     getAllTextDocuments: () => Promise<TextDocument[]>
     getWorkspaceFolder: (uri: string) => WorkspaceFolder | null | undefined
     fs: {
-        copy: (src: string, dest: string) => Promise<void>
+        copyFileWithCreateDir: (src: string, dest: string) => Promise<void>
         exists: (path: string) => Promise<boolean>
         getFileSize: (path: string) => Promise<{ size: number }>
         getServerDataDirPath: (serverName: string) => string
         getTempDirPath: () => string
+        /**
+         * Reads the contents of a directory.
+         * @param {string} path - The path to the directory.
+         * @returns A promise that resolves to an array of Dirent objects.
+         */
         readdir: (path: string) => Promise<Dirent[]>
+        /**
+         * Reads the entire contents of a file.
+         * @param {string} path - The path to the file.
+         * @returns A promise that resolves to the contents of the file as a string.
+         */
```

Comment:
Also, comment should mention that we set default encoding.

Reply:
`Would it make sense to expose options for this and set 'utf-8' as default` 

Sounds good to me

Reply:
I added the `options` param with `encoding` set to 'utf-8' in the standalone implementation. I don't think it is possible to add a default parameter in the interface. Should we mention it in the interface documentation in that case?

If we were to add `options` as a default parameter later, would we need to change all the usages?

Reply:
Definitely document the default encoding and in comments make it clear that there is no way to get a Buffer back (yes, the return type declaration shows that, but really call it out in comments/docs). 

The NodeJS fs/promises readFile returns a Buffer by default and can support a Buffer if null is passed in for the encoding.  The implementation of Flare's readFile in standalone doesn't allow for Buffers (only returns strings) and the default encoding has changed.  This will be unexpected by implementors familiar with NodeJS.

I think this falls under the AppSec advice to document any behavioral changes.

Reply:
Currently the default value for encoding is mentioned in the method's description. The return type is also specified in the description, additionally to function declaration. In my opinion it should be clear, do you think we need to emphasize it even more?